src/ui/widgets/FormulaWidgetUI.js

```diff
@@ -62,11 +62,9 @@ function FormulaWidgetUI(props) {
     } else {
       setValue(data);
     }
-
+    // eslint-disable-next-line react-hooks/exhaustive-deps
     return () => cancelAnimationFrame(requestRef.current);
-
-    // eslint-disable-next-line
-  }, [data, setValue]);
+  }, [data, prevValue, setValue]);
```

Comment:
In order to avoid the 'delayed animation' we were talking:
Passing `prevValue` to the dependencies array causes the delay, we could use same technique as my last comment, using `useRef`
```javascript
const referencedPrevValue = useRef(prevValue);
```

Reply:
Fixed [here](https://github.com/CartoDB/carto-react-lib/commit/75b80e5eb885bc85b650cbae12354641c6e5636f)